### DISCUSSION TITLE
Not able to capture variable through before hook

### DISCUSSION THREAD

**Discussion by amank575** [artilleryio/artillery#1640] [Artillery]
Hi hope you are doing well . So, I am new to community and wanted to perform Load testing on internal API of my organization. 
We use auth-token as authorization which have 15 min time limit and maintain browser session for that particular user . If he logs in with other browser their previous logged in session will self destruct .

I tried before hook to pass my auth-token to all of my VU but it shows invalid token, but when chained, its working fine for 1 VU
before hook-
```diff
config:
  target: https://xxx.xxx.x.x
  tls:
    rejectUnauthorized: false
  before:
    flow: # flow for getting auth token
      - post:
          url : /userDetails/login
          json:
            email: xxx.ai
            password: xxxx
            pass_len: xxx
            info : xxx
          capture:
            - json: "$.token"
              as: "auth_token"
          

scenarios:
  - flow:    
    - post:
        url : /xxxx/xxx
        headers: 
          auth-token: '{{ auth_token }}'
        json:
          data: xxxxx
        expect:
          - statusCode: 200

```
response in debug
![image](https://user-images.githubusercontent.com/29748790/199166976-63a93131-b669-4645-adf4-e06f1a0906f5.png)

When chained
```diff
config:
  target:  https://xxx.xxx.x.x 
 # phases:
 #   - duration: 120
  #    arrivalRate: 10
  #http:
  #  timeout: 120
  tls:
    rejectUnauthorized: false

scenarios:
  - flow :
    - post:
        url : /userDetails/login
        json:
          email: xxx.ai
          password: xxx
          pass_len: xx
          info : xxx
        capture:
          - json: "$.token"
            as: auth_token
        expect:
          - statusCode: 200
    - think: 1 
    - post: 
        url : xxx/xxx
        headers: 
          auth-token: '{{ auth_token }}'
        json:
          data: xxxxxx
        expect:
          - statusCode: 200
```
summary while API were chained
![image](https://user-images.githubusercontent.com/29748790/199167665-b4dec490-7da4-491e-96fe-b1f72127b9f3.png)

**Comment by hassy** [comment]
hi @amank575 👋 `before` needs to be at the top level, rather than nested under `config`.

Working example here:

https://play.artillery.io/?s=2230422e-c967-47ef-ae13-5b5147dec2f2

**Reply by amank575** [reply]
thanks @hassy able to use capture variable
